src/components/split_button/split_button.tsx

```diff
@@ -0,0 +1,225 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, { ReactNode, useState } from 'react';
+import classNames from 'classnames';
+
+import { CommonProps, ExclusiveUnion } from '../common';
+
+import { OuiScreenReaderOnly } from '../accessibility';
+import {
+  OuiSplitButtonColor,
+  OuiSplitButtonControl,
+  OuiSplitButtonControlProps,
+} from './split_button_control';
+import { OuiPopover } from '../popover';
+import { OuiContextMenuItem, OuiContextMenuPanel } from '../context_menu';
+import { keys } from '../../services';
+import { OuiI18n } from '../i18n';
+import { OuiButtonProps } from '../button';
+import { OuiText, OuiTextProps } from '../text';
+
+export interface OuiSplitButtonOption {
+  display: string | ReactNode;
+  onClick?: () => void;
+  href?: string;
+  target?: string;
+}
+
+export type OuiSplitButtonProps = CommonProps &
+  Omit<
+    OuiSplitButtonControlProps,
+    'onKeyDown' | 'onChange' | 'onDropdownClick' | 'options' | 'value'
+  > & {
+    /**
+     * Pass an array of options
+     */
+    options?: OuiSplitButtonOption[];
+
+    /**
+     * Classes for the context menu item
+     */
+    itemClassName?: string;
+
+    /**
+     * optional index of options item to mark with checkmark
+     */
+    selectedIndex?: number;
+
+    /**
+     * This callback will be executed when an option is clicked, in addition to possible
+     * href/onClick of the option-item itself.
+     */
+    onSelection?: (index: number) => void;
+
+    /**
+     * Change to `true` if you want horizontal lines between options.
+     * This is best used when options are multi-line.
+     */
+    hasDividers?: boolean;
+
+    /**
+     * Applied to the outermost wrapper (popover)
+     */
+    popoverClassName?: string;
+
+    /**
+     * Controls whether the options are shown upon initial render. Default: false
+     */
+    initiallyOpen?: boolean;
+
+    /**
+     * Optional additional props to send to Primary Button
+     */
+    propsForPrimaryButton?: OuiButtonProps;
+
+    /**
+     * Optional additional props to send to Dropdown Button
+     */
+    propsForDropdownButton?: OuiButtonProps;
+
+    /**
+     * Optional additional props to send to each Option Item, when
+     * it is string, rendered with OuiText wrapper
+     */
+    propsForOptionItems?: OuiTextProps;
```

Comment:
nit: Can we name this similar to `textProps` for `Button`.

```suggestion
    /**
     * Optional additional props to send to Primary Button
     */
    buttonProps?: OuiButtonProps;

    /**
     * Optional additional props to send to Dropdown Button
     */
    dropdownProps?: OuiButtonProps;

    /**
     * Optional additional props to send to each Option Item, when
     * it is string, rendered with OuiText wrapper
     */
    optionProps?: OuiTextProps;
```